tests/foreman/performance/test_pulp_sync.py

```diff
@@ -0,0 +1,281 @@
+    cached data. The process is repeated X times and return
+    timing values for each thread/repository. Then it would
+    continue test case of 3, 4, ..., N repositories by the
+    above process.
+
+    Resync test, on the other hand, would not restore db
+    between each iteration of a test case. However, it'd
+    still restore and sync all target repos sequentially
+    once, prior to the test case starts. For 2-repo test
+    case, both are synced for first time. Then it starts
+    the resync by waiting the longest to finish, and
+    immediately going to next iteration without reset
+    (also repeated X times in this test case). It still
+    restores and syncs for first time before 3, 4, ..., N
+    test case starts.
+
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(ConcurrentSyncTestCase, cls).setUpClass()
+
+        # note: may need to change savepoint in config file
+        cls._set_testcase_parameters(
+            'performance.test.savepoint2_enabled_repos',
+            RAW_SYNC_FILE_NAME,
+            STAT_SYNC_FILE_NAME,
+        )
+
+        # get enabled repositories information
+        cls.map_repo_name_id = Pulp.get_enabled_repos(cls.org_id)
+        cls.logger.debug(cls.map_repo_name_id)
+
+        # get number of iterations of syncs that each thread would do
+        cls.sync_iterations = int(conf.properties.get(
+            'performance.test.num_syncs',
+            '3'
+        ))
+
+        # get whether start initial sync or resync test
+        sync_parameter = conf.properties.get(
+            'performance.test.is_initial_sync',
+            'sync'
+        )
+        cls.is_initial_sync = True if sync_parameter == 'sync' else False
+
+    def setUp(self):
+        super(ConcurrentSyncTestCase, self).setUp()
+
+        # determine all targeting repositories to be synced
+        self.repo_names_list = self._get_target_repo_from_config()
+        self.logger.debug(
+            "Target Repositories to be synced: {}"
+            .format(self.repo_names_list)
+        )
+
+        self.max_num_tests = len(self.repo_names_list)
+        self.logger.debug(
+            'Max number of repositories to sync: {}'
+            .format(self.max_num_tests)
+        )
+
+    def _get_target_repo_from_config(self):
+        """Read targeting repository names from config file
+
+        note: Targeting repository should be enabled at first
+
+        """
+        repos_str = conf.properties.get('performance.test.target_repos', '')
+        if repos_str == '':
+            raise RuntimeError('Invalid configuration of repositories')
+        repo_names_list = [
+            repo
+            for repo in repos_str.split(',')
+            if repo in self.map_repo_name_id
+        ]
+        return repo_names_list
+
+    def test_concurrent_synchronization(self):
+        """
+        @Test: Synchronize two repos concurrently
+
+        @Steps:
+        1. get list of all enabled repositories (setUpClass)
```

Comment:
Lists should be separated from other paragraphs with a blank line. Just add one line above the `1. get list of all ...` like so:

    @Steps:

    1. get list of all enabled repositories (setUpClass)